crates/chainio/clients/avsregistry/src/reader.rs

```diff
@@ -355,71 +353,77 @@ impl AvsRegistryChainReader {
-                    }
-                    let filter = Filter::new()
-                        .select(i..to_block)
-                        .event("NewPubkeyRegistration(address,(uint256,uint256),(uint256[2],uint256[2]))")
-                        .address(self.bls_apk_registry_addr);
-
-                    let logs_result = provider.get_logs(&filter).await;
-
-                    match logs_result {
-                        Ok(logs) => {
-                            debug!(transactionLogs = ?logs, "avsRegistryChainReader.QueryExistingRegisteredOperatorPubKeys");
-
-                            for v_log in logs.iter() {
-                                if let Ok(pub_key_reg) =
-                                    v_log.log_decode::<BLSApkRegistry::NewPubkeyRegistration>()
-                                {
-                                    let data = pub_key_reg.data();
-                                    let operator_addr = data.operator;
-                                    operator_addresses.push(operator_addr);
-                                    let g1_pub_key = data.pubkeyG1.clone();
-                                    let g2_pub_key = data.pubkeyG2.clone();
-
-                                    let operator_pub_key = OperatorPubKeys {
-                                        g1_pub_key,
-                                        g2_pub_key,
-                                    };
-
-                                    operator_pub_keys.push(operator_pub_key);
+                match current_block_number_result {
+                    Ok(current_block_number) => {
+                        if stop_block.is_zero() {
+                            stop_block = current_block_number;
+                        }
+                        let mut i = start_block;
+                        let mut operator_addresses: Vec<Address> = vec![];
+                        let mut operator_pub_keys: Vec<OperatorPubKeys> = vec![];
+                        while i <= stop_block {
+                            let mut to_block = i + (query_block_range - 1);
+                            if to_block > stop_block {
+                                to_block = stop_block;
+                            }
+                            let filter = Filter::new()
+                                .select(i..to_block)
+                                .event("NewPubkeyRegistration(address,(uint256,uint256),(uint256[2],uint256[2]))")
+                                .address(self.bls_apk_registry_addr);
+
+                            let logs_result = provider.get_logs(&filter).await;
+
+                            match logs_result {
+                                Ok(logs) => {
+                                    debug!(transactionLogs = ?logs, "avsRegistryChainReader.QueryExistingRegisteredOperatorPubKeys");
+
+                                    for v_log in logs.iter() {
+                                        if let Ok(pub_key_reg) =
```

Comment:
what happens on Err, should we ignore or log? at least leave a comment

if we're ignore Err then you can do:
```
for v_log in logs.iter().map(|v| v.log_decode::<BLSApkRegistry::NewPubkeyRegistration>()).filter(|v| v.is_some())
```

and avoid the inner if